hcloud/actions/client.py

```diff
@@ -1,12 +1,21 @@
 # -*- coding: utf-8 -*-
-from hcloud.core.client import ClientEntityBase, BoundModelBase
+import time
 
-from hcloud.actions.domain import Action
+from hcloud.core.client import ClientEntityBase, BoundModelBase
+from hcloud.actions.domain import Action, ActionFailedException
 
 
 class BoundAction(BoundModelBase):
     model = Action
 
+    def wait_until_finished(self):
+        from hcloud import HcloudClient
+        while self.status == self.STATE_RUNNING:
+            self.reload()
+            time.sleep(HcloudClient.poll_interval)
+        if self.status == self.STATE_ERROR:
```

Comment:
It is a misusing of `BoundModelBase.__getattr__` logic. Please refer to `STATE_ERROR` directly (`Action.STATE_ERROR`)